src/simulators/statevector/qubitvector.hpp

```diff
@@ -516,128 +684,120 @@ cvector_t QubitVector<statevector_t>::vector() const {
 // Utility
 //------------------------------------------------------------------------------
 
-template <class statevector_t>
-void QubitVector<statevector_t>::initialize() {
-  const int_t end = num_states_;    // end for k loop
-
-#pragma omp parallel for if (num_qubits_ > omp_threshold_ && omp_threads_ > 1) num_threads(omp_threads_)
-  for (int_t k = 0; k < end; ++k)
-    statevector_[k] = 0.0;
-
-  statevector_[0] = 1.;
-}
-
-template <class statevector_t>
-void QubitVector<statevector_t>::initialize(const cvector_t &statevec) {
-  if (num_states_ != statevec.size()) {
-    std::stringstream ss;
-    ss << "QubitVector<statevector_t>::initialize input vector is incorrect length (";
-    ss << num_states_ << "!=" << statevec.size() << ")";
-    throw std::runtime_error(ss.str());
-  }
-
-  const int_t end = num_states_;    // end for k loop
+template <class data_t>
+void QubitVector<data_t>::zero() {
+  const int_t end = size_;    // end for k loop
 
 #pragma omp parallel for if (num_qubits_ > omp_threshold_ && omp_threads_ > 1) num_threads(omp_threads_)
-  for (int_t k = 0; k < end; ++k)
-    statevector_[k] = statevec[k];
-}
-
-template <class statevector_t>
-void QubitVector<statevector_t>::initialize(const statevector_t &statevec, const size_t num_states) {
-  if (num_states_ != num_states) {
-    std::stringstream ss;
-    ss << "QubitVector<statevector_t>::initialize input vector is incorrect length (";
-    ss << num_states_ << "!=" << num_states << ")";
-    throw std::runtime_error(ss.str());
+  for (int_t k = 0; k < end; ++k) {
+    data_[k] = 0.0;
   }
-
-  const int_t end = num_states_;    // end for k loop
-
-#pragma omp parallel for if (num_qubits_ > omp_threshold_ && omp_threads_ > 1) num_threads(omp_threads_)
-  for (int_t k = 0; k < end; ++k)
-    statevector_[k] = statevec[k];
 }
 
-template <class statevector_t>
-void QubitVector<statevector_t>::set_num_qubits(size_t num_qubits) {
+template <class data_t>
+void QubitVector<data_t>::set_num_qubits(size_t num_qubits) {
   num_qubits_ = num_qubits;
-  num_states_ = 1ULL << num_qubits;
+  size_ = 1ULL << num_qubits;
 
   // Free any currently assigned memory
-  if (statevector_)
-    free(statevector_);
+  if (data_)
+    free(data_);
 
   if (checkpoint_) {
     free(checkpoint_);
     checkpoint_ = 0;
   }
 
   // Allocate memory for new vector
-  statevector_ = reinterpret_cast<complex_t*>(malloc(sizeof(complex_t) * num_states_));
+  data_ = reinterpret_cast<complex_t*>(malloc(sizeof(complex_t) * size_));
```

Comment:
ok, I think this is coming from one of the Hiroshi's optimizations... but can't really remember why we are using a plain `malloc` instead of a ` new`, which is the idiomatic way of allocating heap memory in C++

Reply:
@hhorii can we change this  `malloc` into a `new`?  Is there any known performance degradation with this change?

Reply:
I think Hiroshi tested performance of the different ways and this was the fastest